corgidrp/polflat.py

```diff
@@ -0,0 +1,377 @@
+            dataset (corgidrp.data.Dataset): a dataset of image frames that are raster scanned (L2a-level)
+            nr (int): planet radius
+            pfov_size (int): size of the image created for the polarization variation, typically ~ 2 x nr
+            image_center_x (int): x coordinate of the center pixel of the final image with two orthogonal pol components
+            image_center_y (int): y coordinate of the center pixel of the final image with two orthogonal pol components
+            separation_diameter_arcsec (float): separation between the two orthogonal pol components in arcsecs
+            alignment_angle_WP1 (int): wollaston prism angle for Pol0 - 0
+            alignment_angle_WP2 (int): wollaston prism angle for Pol45- 45
+            planet (str): neptune or uranus
+            band (str): 1 or 4
+            prism (str): wollaston prism pol0 or pol45
+            
+    	Returns:
+    		data.Dataset: dataset of uranus or neptune with spatial variation of polarization corresponding to specific wollaston prism
+    		
+	"""
+    pfov = pfov_size
+    polar_fov = np.ones((pfov,pfov))
+
+     # default to unvignetted polarimetry FOV diameter of 3.8"
+    radius_arcsec = 3.8 / 2
+    # convert to pixel: 0.0218" = 1 pixel
+    radius_pix = int(round(radius_arcsec / 0.0218))
+
+    x = np.arange(0,pfov)
+    y = np.arange(0,pfov)
+    xx, yy = np.meshgrid(x,y)
+    nrr = np.sqrt((xx-(pfov//2))**2 + (yy-(pfov//2))**2)
+
+    polar_filter = np.logical_and(yy<.99*xx,yy<.99*(pfov-xx)) + np.logical_and(yy>.99*xx,yy>.99*(pfov-xx))
+    
+    for i in range(len(dataset)):
+        target=dataset[i].pri_hdr['TARGET']
+        filter=dataset[i].pri_hdr['FILTER']
+        if planet==target and band==filter: 
+            planet_image=dataset[i].data
+    
+    if planet == 'uranus' and band=="1":
+        polar_fov[polar_filter==True] = .0115
+        polar_fov[polar_filter==False] = -0.0115
+    elif planet == 'uranus' and band=="4" or planet == 'neptune' and band=="4":
+        polar_fov[polar_filter==True] = .005
+        polar_fov[polar_filter==False] = -0.005
+    elif planet == 'neptune' and band=="1":
+        polar_fov[polar_filter==True] = .006
+        polar_fov[polar_filter==False] = -0.006
+    r_xy = polar_fov
+    
+    n_rad=nr
+    if n_rad is None:
+        if planet.lower() =='neptune':
+             n_rad = 60
+        elif planet.lower() == 'uranus':
+             n_rad = 95
+    
+    r_xy[nrr>=n_rad] = 0
+    
+   
+    u_data=planet_image
+    I_1 = u_data.copy()
+    I_2 = u_data.copy()
+    centroid_init = centr.centroid_1dg(u_data)
+    xc_init=int(centroid_init[0])
+    yc_init=int(centroid_init[1])
+
+
+    if prism == 'POL0':
+    #place image according to specified angle
+        angle_rad = (alignment_angle_WP1 * np.pi) / 180
+    else:
+        angle_rad = (alignment_angle_WP2 * np.pi) / 180
+     
+    
+    displacement_x = int(round((separation_diameter_arcsec * np.cos(angle_rad)) / (2 * 0.0218)))
```

Comment:
we're going to be measuring the real platescale on-sky with the Astrometric Calibration. Should we be passing in the astrometric calibration and getting the platesclae from it, instead of hard coding it to an approximate number?